libpromises/cmdb.c

```diff
@@ -105,13 +105,33 @@ static bool ReadCMDBVars(EvalContext *ctx, JsonElement *vars)
     {
         const char *key = JsonIteratorNextKey(&iter);
         JsonElement *data = JsonObjectGet(vars, key);
-        VarRef ref = VarRefConst("cmdb", "variables", key);
+
+        VarRef *ref = VarRefParse(key);
+        if (ref->ns == NULL)
+        {
+            ref->ns = xstrdup("cmdb");
+        }
+        else
+        {
+            if (ref->scope == NULL)
+            {
+                Log(LOG_LEVEL_ERR, "Invalid variable specification in CMDB data: '%s'"
+                    " (bundle name has to be specified if namespace is specified)", key);
+                VarRefDestroy(ref);
+                continue;
+            }
+        }
+
+        if (ref->scope == NULL)
+        {
+            ref->scope = xstrdup("variables");
+        }
 
         if (JsonGetElementType(data) == JSON_ELEMENT_TYPE_PRIMITIVE)
         {
             char *value = JsonPrimitiveToString(data);
             Log(LOG_LEVEL_VERBOSE, "Installing CMDB variable 'cmdb:variables.%s=%s'", key, value);
-            EvalContextVariablePut(ctx, &ref, value, CF_DATA_TYPE_STRING, "source=cmdb");
+            EvalContextVariablePut(ctx, ref, value, CF_DATA_TYPE_STRING, "source=cmdb");
```

Comment:
This tag, `source=cmdb`, is applied under what conditions? Is it the use of the new `variables` key, or is it the location of the source data file?

Reply:
No, this is just the default for `vars` variables that cannot have specific tags.